plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -28,17 +44,97 @@
  */
 public class GoogleLoginService {
 
-  // TODO(chanseok): this will be needed later.
-  public static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
-      "https://www.googleapis.com/auth/userinfo#email"
+  private static final String STASH_OAUTH_CRED_KEY = "OAUTH_CRED";
+  private static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
+      "email", //$NON-NLS-1$
+      "https://www.googleapis.com/auth/cloud-platform" //$NON-NLS-1$
   ));
 
   /**
-   * Returns the credential of the active user. If there is no active user, returns {@code null}.
+   * Returns the credential of an active user (among multiple logged-in users). A login screen
+   * may be presented, e.g., if no user is logged in or login is required due to an expired
+   * credential.
+   *
+   * @param shellProvider provides a shell for the login screen if login is necessary
+   * @throws IOException can be thrown by the underlying Login API request
+   */
+  // TODO(chanseok); synchronize properly
+  public Credential getActiveCredential(IShellProvider shellProvider) throws IOException {
+    MApplication app = PlatformUI.getWorkbench().getService(MApplication.class);
+    Credential credential = (Credential) app.getTransientData().get(STASH_OAUTH_CRED_KEY);
```

Comment:
I think this looks fine.  Going in the future, it might be better to turn `GoogleLoginService` into an OSGi-level or Workbench-level service — just as you're fetching the `MApplication` here.  The OSGi Services and Eclipse Contexts (`IEclipseContext`) provide a nifty way to access and store information simplifying the maintenance of per-window and per-application state.